### PR TITLE
Added support for detecting premium domains via urn:ar:params:xml:ns:price-1.2

### PR DESCRIPTION
**Pull request by wbond**

### CODE REVIEW

File: check.go [hunk 1]
```diff
@@ -149,4 +190,26 @@ func init() {
...
+	scanResponse.MustHandleCharData(path+">cd>name", func(c *xx.Context) error {
+		charges := c.Value.(*response_).DomainCheckResponse.Charges
+		charge := &charges[len(charges)-1]
+		charge.Domain = string(c.CharData)
+		return nil
+	})
+	scanResponse.MustHandleCharData(path+">cd>category", func(c *xx.Context) error {
+		charges := c.Value.(*response_).DomainCheckResponse.Charges
+		charge := &charges[len(charges)-1]
+		if strings.Index(strings.ToLower(string(c.CharData)), "premium") != -1 {
```
Comment (ydnar) [inline check.go]: Maybe use `bytes.Contains(c.CharData, pricePremium)`?

Where `pricePremium = []byte("remium")`?

**Comment by ydnar** [comment]
@wbond what happens when you pass the ARI price extension to the first check command?

**Comment by wbond** [comment]
No availability info is returned. :-/ That is why I had to add a second call.

File: check.go (hunk 1 above)
Comment (wbond) [inline check.go]: It seems like the categories aren't very standardized. For instance, a regular domain is `STANDARD`, while the few premiums I saw used `Premium_` with numbers added.

I realize the code is more complicated, I figured it was worth making it more robust.

**Comment by ydnar** [comment]
OK. Have we positively confirmed that two calls are necessary with ARI?

If so, then merge away.

**Comment by case** [comment]
I'll confirm with ARI.

**Comment by case** [comment]
Oddly, they directed me [here](http://ausregistry.github.io/doc/exAvail-1.0/exAvail-1.0.html), which doesn't appear to mention premiums.

Poking around further, the v1.0 and v1.1 price extensions appear to support availability and premium status in a single call:

- http://ausregistry.github.io/doc/price-1.0/price-1.0.html
- http://ausregistry.github.io/doc/price-1.1/price-1.1.html

But for some reason, the v1.2 extension doesn't mention premiums:

- http://ausregistry.github.io/doc/price-1.2/price-1.2.html

I'm checking with them to see if 1.0 and 1.1 will be deprecated at some point, or if we can safely use them.

**Comment by case** [comment]
Confirmed with ARI that we can use v1.0 or v1.1: "I have checked with our product manager, there are no immediate plans to deprecate v1.0 or v1.1 , each version has been developed to address Registry Operator requirements for different zones."

**Comment by wbond** [comment]
If you look at the XML returned by the [price-1.0](http://ausregistry.github.io/doc/price-1.0/price-1.0.html#anchor7) and [price-1.1](http://ausregistry.github.io/doc/price-1.1/price-1.1.html#anchor7) docs, you can see that no availability information is returned when any of the price extensions are specified in the request. For comparison, here is a `<resData>` tag returned by an availability check: https://github.com/domainr/epp/blob/master/check_test.go#L109-L113.

I've tried using both price 1.0 and 1.1, and confirmed the `<resData>` tag is not present in the response from either. Since it is not included, we can't get around making two EPP requests.

However, we could use price 1.0 or 1.1 to get a simpler method to detect if a domain is premium, since both versions expose a `premium="1"` attribute on the `<name>` tag that is returned. With price 1.2, there is a category name returned, which I was having to use string matching to detect the value `premium`.

**Comment by case** [comment]
Bummer. The categories in 1.2 might actually be more useful, e.g. there may be one for `reserved` names, etc.

@ydnar it sounds like we've hit a wall here — ARI apparently doesn't have a single call that returns both availability status and premium status.

**Comment by ydnar** [comment]
@wbond if you pass the extension request and the normal check request XML in the same check XML request, does it work?

**Comment by wbond** [comment]
@ydnar The ARI EPP server response with a `Command syntax error` if we try either:

```xml
<command>
    <check></check>
    <check></check> 
    <extension></extension>
</command>
```

or

```xml
<command>
    <check></check>
</command>
<command>
    <check></check>
    <extension></extension>
</command>
```

This seems to jive with [RFC5730](https://tools.ietf.org/html/rfc5730#section-2.5).

**Pull request merged**